src/utils/Stake.ts

```diff
@@ -15,6 +16,13 @@ export function createOrUpdateStake(event: DelegateStakeChanged): void {
   stake.amount = decimal.fromBigInt(event.params.newBalance, DEFAULT_DECIMALS)
   stake.lockedUntil = event.params.lockedUntil.toI32()
   stake.save()
+  return stake
+}
+
+export function removeStakeIfEmpty(stake: Stake): void {
+  if (stake != null && (!stake.amount || (stake.amount as BigDecimal).le(BigDecimal.zero()))) {
```

Comment:
I don't think you need this null check - stake will never be null (if it could be it would not compile).

Also stake.amount is already a BigDecimal so you don't need to type cast it

Reply:
Hm, I see this in the generated schema typings, and subgraph does not build without these checks and casting

<img width="372" alt="Screenshot 2023-04-26 at 16 55 23" src="https://user-images.githubusercontent.com/69458664/234598138-1b1141c6-467f-4d24-a882-71d0fd648f28.png">

Reply:
If it was supposed to be non-nullable, I will change entity schema to make sure amount is required

Reply:
I don't mean the null check on amount, I meant null check on stake